modules/mobile-agents/android/jax-rs/src/main/java/org/wso2/carbon/mdm/services/android/PolicyMgtService.java

```diff
@@ -43,20 +42,31 @@
 @Produces({"application/json", "application/xml"})
 @Consumes({"application/json", "application/xml"})
 public class PolicyMgtService {
-    private static Log log = LogFactory.getLog(PolicyMgtService.class);
 
-    @POST
-    @Path("/getpolicy")
-    public Policy getEffectivePolicy(DeviceIdentifier deviceIdentifier) throws AndroidAgentException {
+	private static Log log = LogFactory.getLog(PolicyMgtService.class);
+
+    @GET
+    @Path("{id}")
+    public Message getEffectivePolicy(@HeaderParam("Accept") String acceptHeader,
+									   @PathParam("id") String id) throws AndroidAgentException {
+
+		DeviceIdentifier deviceIdentifier = AndroidAPIUtils.convertToDeviceIdentifierObject(id);
+		MediaType responseMediaType = AndroidAPIUtils.getResponseMediaType(acceptHeader);
+		Message responseMessage = new Message();
         Policy policy;
         try {
-
             PolicyManagerService policyManagerService = AndroidAPIUtils.getPolicyManagerService();
             policy = policyManagerService.getEffectivePolicy(deviceIdentifier);
             if (policy == null) {
-                Response.status(Response.Status.NOT_FOUND);
-            }
-            return policy;
+				responseMessage.setResponseCode(Response.Status.NO_CONTENT.toString());
+				responseMessage.setResponseMessage("No effective policy found");
+                return  responseMessage;
+            } else {
+				responseMessage.setResponseCode(Response.Status.OK.toString());
+				responseMessage.setResponseMessage("Effective policy added to operation");
+				return  responseMessage;
+			}
+
         } catch (PolicyManagementException e) {
             String msg = "Error occurred while getting the policy.";
```

Comment:
Creation of the msg variable is not necessary here since you are using it only once.